Bug: `for` loop does not allow optional parentheses (inconsistent with other control flow)

## Summary

`for` loops fail when parentheses are used around the loop expression, while `as_long_as`, `if`, and `or` all accept optional parentheses.

## Error

```
error[E2002]: expected IDENT, got ( instead
```

## Reproduction

```ez
// This FAILS
for (i in range(0, 3)) {
    println(i)
}

// This WORKS
for i in range(0, 3) {
    println(i)
}
```

## Inconsistency

```ez
// as_long_as - parens optional (BOTH work)
as_long_as (x < 10) { }
as_long_as x < 10 { }

// if/or - parens optional (BOTH work)
if (x > 5) { } or (x < 0) { }
if x > 5 { } or x < 0 { }

// for - parens NOT allowed (inconsistent)
for (i in range(0, 3)) { }  // ERROR
for i in range(0, 3) { }    // OK

// loop - parens not allowed (correct - no condition)
loop { }  // OK
```

## Expected Behavior

Allow optional parentheses in `for` loops to be consistent with `as_long_as`, `if`, and `or`.

## Notes

- `loop` correctly rejects parentheses since it has no condition
- Only `for` is inconsistent with the other conditional statements

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -1007,6 +1007,12 @@ func (p *Parser) parseAlternative() Statement {
 func (p *Parser) parseForStatement() *ForStatement {
 	stmt := &ForStatement{Token: p.currentToken}
 
+	// Check for optional opening parenthesis
+	hasParens := p.peekTokenMatches(LPAREN)
+	if hasParens {
+		p.nextToken() // consume '('
+	}
+
 	if !p.expectPeek(IDENT) {
 		return nil
 	}
@@ -1026,6 +1032,13 @@ func (p *Parser) parseForStatement() *ForStatement {
 	p.nextToken() // move past 'in'
 	stmt.Iterable = p.parseExpression(LOWEST)
 
+	// Check for optional closing parenthesis
+	if hasParens {
+		if !p.expectPeek(RPAREN) {
+			return nil
+		}
+	}
+
 	if !p.expectPeek(LBRACE) {
 		return nil
 	}
@@ -1037,6 +1050,12 @@ func (p *Parser) parseForStatement() *ForStatement {
 
 func (p *Parser) parseForEachStatement() *ForEachStatement {
 	stmt := &ForEachStatement{Token: p.currentToken}
+
+	// Check for optional opening parenthesis
+	hasParens := p.peekTokenMatches(LPAREN)
+	if hasParens {
+		p.nextToken() // consume '('
+	}
 
 	if !p.expectPeek(IDENT) {
 		return nil
@@ -1050,6 +1069,13 @@ func (p *Parser) parseForEachStatement() *ForEachStatement {
 
 	p.nextToken() // move past 'in'
 	stmt.Collection = p.parseExpression(LOWEST)
+
+	// Check for optional closing parenthesis
+	if hasParens {
+		if !p.expectPeek(RPAREN) {
+			return nil
+		}
+	}
 
 	if !p.expectPeek(LBRACE) {
 		return nil

--- a/pkg/parser/parser_test.go
+++ b/pkg/parser/parser_test.go
@@ -880,6 +880,237 @@ func TestForEachStatement(t *testing.T) {
 	}
 }
 
+func TestForStatementWithParens(t *testing.T) {
+	// Test optional parentheses around for loop expression
+	input := `for (i in range(0, 10)) {
+		x = i
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*ForStatement)
+
+	if stmt.Variable == nil {
+		t.Error("variable is nil")
+	}
+	if stmt.Variable.Value != "i" {
+		t.Errorf("expected variable 'i', got %q", stmt.Variable.Value)
+	}
+	if stmt.Iterable == nil {
+		t.Error("iterable is nil")
+	}
+}
+
+func TestForStatementWithParensAndType(t *testing.T) {
+	// Test optional parentheses with type annotation
+	input := `for (i int in range(0, 10)) {
+		x = i
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*ForStatement)
+
+	if stmt.Variable.Value != "i" {
+		t.Errorf("expected variable 'i', got %q", stmt.Variable.Value)
+	}
+	if stmt.VarType != "int" {
+		t.Errorf("expected VarType 'int', got %q", stmt.VarType)
+	}
+}
+
+func TestForEachStatementWithParens(t *testing.T) {
+	// Test optional parentheses around for_each loop expression
+	input := `for_each (item in items) {
+		process(item)
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*ForEachStatement)
+
+	if stmt.Variable == nil {
+		t.Fatal("variable is nil")
+	}
+	if stmt.Variable.Value != "item" {
+		t.Errorf("expected variable 'item', got %q", stmt.Variable.Value)
+	}
+	if stmt.Collection == nil {
+		t.Error("collection is nil")
+	}
+}
+
+func TestForStatementNestedWithParens(t *testing.T) {
+	// Test nested for loops both with parentheses
+	input := `for (i in range(0, 3)) {
+		for (j in range(0, 2)) {
+			x = i + j
+		}
+	}`
+	program := parseProgram(t, input)
+	outerFor := program.Statements[0].(*ForStatement)
+
+	if outerFor.Variable.Value != "i" {
+		t.Errorf("expected outer variable 'i', got %q", outerFor.Variable.Value)
+	}
+
+	// Check inner for loop
+	if len(outerFor.Body.Statements) == 0 {
+		t.Fatal("outer body is empty")
+	}
+	innerFor, ok := outerFor.Body.Statements[0].(*ForStatement)
+	if !ok {
+		t.Fatalf("expected inner ForStatement, got %T", outerFor.Body.Statements[0])
+	}
+	if innerFor.Variable.Value != "j" {
+		t.Errorf("expected inner variable 'j', got %q", innerFor.Variable.Value)
+	}
+}
+
+func TestForStatementWithComplexIterable(t *testing.T) {
+	// Test for loop with parentheses containing function call
+	input := `for (i in get_range()) {
+		x = i
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*ForStatement)
+
+	if stmt.Variable.Value != "i" {
+		t.Errorf("expected variable 'i', got %q", stmt.Variable.Value)
+	}
+	// Verify iterable is a call expression
+	_, ok := stmt.Iterable.(*CallExpression)
+	if !ok {
+		t.Errorf("expected iterable to be CallExpression, got %T", stmt.Iterable)
+	}
+}
+
+func TestForStatementWithArrayLiteralIterable(t *testing.T) {
+	// Test for loop with parentheses containing array literal
+	input := `for (x in {1, 2, 3}) {
+		sum += x
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*ForStatement)
+
+	if stmt.Variable.Value != "x" {
+		t.Errorf("expected variable 'x', got %q", stmt.Variable.Value)
+	}
+	// Verify iterable is an array value
+	_, ok := stmt.Iterable.(*ArrayValue)
+	if !ok {
+		t.Errorf("expected iterable to be ArrayValue, got %T", stmt.Iterable)
+	}
+}
+
+func TestForStatementMixedParensStyle(t *testing.T) {
+	// Test that we can have outer for with parens, inner without
+	input := `for (i in range(0, 3)) {
+		for j in range(0, 2) {
+			x = i + j
+		}
+	}`
+	program := parseProgram(t, input)
+	outerFor := program.Statements[0].(*ForStatement)
+
+	if outerFor.Variable.Value != "i" {
+		t.Errorf("expected outer variable 'i', got %q", outerFor.Variable.Value)
+	}
+
+	innerFor := outerFor.Body.Statements[0].(*ForStatement)
+	if innerFor.Variable.Value != "j" {
+		t.Errorf("expected inner variable 'j', got %q", innerFor.Variable.Value)
+	}
+}
+
+func TestForStatementParensMissingClosing(t *testing.T) {
+	// Test that missing closing paren produces an error
+	input := `for (i in range(0, 10) {
+		x = i
+	}`
+	l := NewLexer(input)
+	p := New(l)
+	p.ParseProgram()
+
+	errs := p.Errors()
+	if len(errs) == 0 {
+		t.Error("expected parser error for missing closing paren, got none")
+	}
+}
+
+func TestForEachStatementParensMissingClosing(t *testing.T) {
+	// Test that missing closing paren produces an error
+	input := `for_each (item in items {
+		process(item)
+	}`
+	l := NewLexer(input)
+	p := New(l)
+	p.ParseProgram()
+
+	errs := p.Errors()
+	if len(errs) == 0 {
+		t.Error("expected parser error for missing closing paren, got none")
+	}
+}
+
+func TestWhileStatementWithParens(t *testing.T) {
+	// Test as_long_as with optional parentheses
+	input := `as_long_as (x < 10) {
+		x++
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*WhileStatement)
+
+	if stmt.Condition == nil {
+		t.Error("condition is nil")
+	}
+}
+
+func TestIfStatementWithParens(t *testing.T) {
+	// Test if with optional parentheses
+	input := `if (x > 5) {
+		y = 1
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*IfStatement)
+
+	if stmt.Condition == nil {
+		t.Error("condition is nil")
+	}
+}
+
+func TestIfOrStatementWithParens(t *testing.T) {
+	// Test if/or/otherwise with optional parentheses
+	input := `if (x < 10) {
+		y = 1
+	} or (x < 20) {
+		y = 2
+	} otherwise {
+		y = 3
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*IfStatement)
+
+	if stmt.Condition == nil {
+		t.Error("if condition is nil")
+	}
+	if stmt.Alternative == nil {
+		t.Error("alternative (or clause) is nil")
+	}
+}
+
+func TestIfOrStatementMixedParens(t *testing.T) {
+	// Test if without parens, or with parens
+	input := `if x < 10 {
+		y = 1
+	} or (x < 20) {
+		y = 2
+	}`
+	program := parseProgram(t, input)
+	stmt := program.Statements[0].(*IfStatement)
+
+	if stmt.Condition == nil {
+		t.Error("if condition is nil")
+	}
+	if stmt.Alternative == nil {
+		t.Error("or clause is nil")
+	}
+}
+
 func TestWhileStatement(t *testing.T) {
 	input := `as_long_as x < 10 {
 		x++

--- a/tests/comprehensive.ez
+++ b/tests/comprehensive.ez
@@ -109,6 +109,10 @@ do main() {
     total_passed += result.passed
     total_failed += result.failed
 
+    result = test_optional_parens()
+    total_passed += result.passed
+    total_failed += result.failed
+
     result = test_arrays()
     total_passed += result.passed
     total_failed += result.failed
@@ -423,6 +427,307 @@ do test_control_flow() -> TestResult {
     }
     println("    sum with skip = ${cont_sum}")
     passed += 1
+
+    println("  PASSED: ${passed}, FAILED: ${failed}")
+    return TestResult{passed: passed, failed: failed}
+}
+
+// ==================================================
+// OPTIONAL PARENTHESES IN CONTROL FLOW
+// ==================================================
+
+do test_optional_parens() -> TestResult {
+    print_section("Optional Parentheses")
+    temp passed int = 0
+    temp failed int = 0
+
+    // ========== FOR LOOPS ==========
+
+    // for loop WITHOUT parentheses (original syntax)
+    println("  for without parens:")
+    temp sum1 int = 0
+    for i in range(0, 5) {
+        sum1 += i
+    }
+    println("    sum = ${sum1}")
+    if sum1 == 10 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 10")
+    }
+
+    // for loop WITH parentheses (new syntax)
+    println("  for with parens:")
+    temp sum2 int = 0
+    for (i in range(0, 5)) {
+        sum2 += i
+    }
+    println("    sum = ${sum2}")
+    if sum2 == 10 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 10")
+    }
+
+    // for loop with type annotation WITHOUT parens
+    println("  for with type, no parens:")
+    temp sum3 int = 0
+    for i int in range(0, 3) {
+        sum3 += i
+    }
+    println("    sum = ${sum3}")
+    if sum3 == 3 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 3")
+    }
+
+    // for loop with type annotation WITH parens
+    println("  for with type and parens:")
+    temp sum4 int = 0
+    for (i int in range(0, 3)) {
+        sum4 += i
+    }
+    println("    sum = ${sum4}")
+    if sum4 == 3 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 3")
+    }
+
+    // for loop with descending range WITH parens
+    println("  for descending with parens:")
+    temp desc_list string = ""
+    for (i in range(3, 0)) {
+        desc_list = "${desc_list}${i}"
+    }
+    println("    result = ${desc_list}")
+    if desc_list == "321" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected '321'")
+    }
+
+    // ========== FOR_EACH LOOPS ==========
+
+    // for_each WITHOUT parentheses
+    println("  for_each without parens:")
+    temp items [string] = {"a", "b", "c"}
+    temp concat1 string = ""
+    for_each item in items {
+        concat1 = "${concat1}${item}"
+    }
+    println("    result = ${concat1}")
+    if concat1 == "abc" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'abc'")
+    }
+
+    // for_each WITH parentheses
+    println("  for_each with parens:")
+    temp concat2 string = ""
+    for_each (item in items) {
+        concat2 = "${concat2}${item}"
+    }
+    println("    result = ${concat2}")
+    if concat2 == "abc" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'abc'")
+    }
+
+    // ========== IF STATEMENTS ==========
+
+    // if WITHOUT parentheses
+    println("  if without parens:")
+    temp x int = 5
+    temp result1 string = ""
+    if x > 3 {
+        result1 = "gt3"
+    }
+    println("    result = ${result1}")
+    if result1 == "gt3" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'gt3'")
+    }
+
+    // if WITH parentheses
+    println("  if with parens:")
+    temp result2 string = ""
+    if (x > 3) {
+        result2 = "gt3"
+    }
+    println("    result = ${result2}")
+    if result2 == "gt3" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'gt3'")
+    }
+
+    // if/or/otherwise WITHOUT parentheses
+    println("  if/or/otherwise without parens:")
+    temp y int = 15
+    temp result3 string = ""
+    if y < 10 {
+        result3 = "small"
+    } or y < 20 {
+        result3 = "medium"
+    } otherwise {
+        result3 = "large"
+    }
+    println("    result = ${result3}")
+    if result3 == "medium" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'medium'")
+    }
+
+    // if/or/otherwise WITH parentheses
+    println("  if/or/otherwise with parens:")
+    temp result4 string = ""
+    if (y < 10) {
+        result4 = "small"
+    } or (y < 20) {
+        result4 = "medium"
+    } otherwise {
+        result4 = "large"
+    }
+    println("    result = ${result4}")
+    if result4 == "medium" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'medium'")
+    }
+
+    // Mixed parens in if/or chain
+    println("  if/or mixed parens:")
+    temp result5 string = ""
+    if y < 10 {
+        result5 = "small"
+    } or (y < 20) {
+        result5 = "medium"
+    } otherwise {
+        result5 = "large"
+    }
+    println("    result = ${result5}")
+    if result5 == "medium" {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 'medium'")
+    }
+
+    // ========== AS_LONG_AS (WHILE) ==========
+
+    // as_long_as WITHOUT parentheses
+    println("  as_long_as without parens:")
+    temp count1 int = 0
+    as_long_as count1 < 3 {
+        count1 += 1
+    }
+    println("    count = ${count1}")
+    if count1 == 3 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 3")
+    }
+
+    // as_long_as WITH parentheses
+    println("  as_long_as with parens:")
+    temp count2 int = 0
+    as_long_as (count2 < 3) {
+        count2 += 1
+    }
+    println("    count = ${count2}")
+    if count2 == 3 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 3")
+    }
+
+    // ========== NESTED CONTROL FLOW WITH PARENS ==========
+
+    // Nested for loops with parens
+    println("  nested for loops with parens:")
+    temp nested_sum int = 0
+    for (i in range(0, 3)) {
+        for (j in range(0, 2)) {
+            nested_sum += 1
+        }
+    }
+    println("    sum = ${nested_sum}")
+    if nested_sum == 6 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 6")
+    }
+
+    // for with if inside, both with parens
+    println("  for with if inside (both parens):")
+    temp even_sum int = 0
+    for (i in range(0, 6)) {
+        if (i % 2 == 0) {
+            even_sum += i
+        }
+    }
+    println("    even sum = ${even_sum}")
+    if even_sum == 6 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 6 (0+2+4)")
+    }
+
+    // as_long_as with break inside, both with parens
+    println("  as_long_as with break (both parens):")
+    temp break_count int = 0
+    as_long_as (break_count < 100) {
+        if (break_count == 5) {
+            break
+        }
+        break_count += 1
+    }
+    println("    count at break = ${break_count}")
+    if break_count == 5 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 5")
+    }
+
+    // for_each with continue, both with parens
+    println("  for_each with continue (both parens):")
+    temp nums [int] = {1, 2, 3, 4, 5}
+    temp skip_sum int = 0
+    for_each (n in nums) {
+        if (n == 3) {
+            continue
+        }
+        skip_sum += n
+    }
+    println("    sum without 3 = ${skip_sum}")
+    if skip_sum == 12 {
+        passed += 1
+    } otherwise {
+        failed += 1
+        println("    FAIL: expected 12 (1+2+4+5)")
+    }
 
     println("  PASSED: ${passed}, FAILED: ${failed}")
     return TestResult{passed: passed, failed: failed}

--- a/tests/errors/comprehensive/E2002_for_each_extra_closing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_each_extra_closing_paren.ez
@@ -1,0 +1,11 @@
+/*
+ * Error Test: E2002 - for_each without opening paren but with closing paren
+ * Expected: "expected {, got ) instead"
+ */
+
+do main() {
+    temp items [int] = {1, 2, 3}
+    for_each item in items) {
+        temp x int = item
+    }
+}

--- a/tests/errors/comprehensive/E2002_for_each_missing_closing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_each_missing_closing_paren.ez
@@ -1,0 +1,11 @@
+/*
+ * Error Test: E2002 - for_each with opening paren but missing closing paren
+ * Expected: "expected ), got { instead"
+ */
+
+do main() {
+    temp items [int] = {1, 2, 3}
+    for_each (item in items {
+        temp x int = item
+    }
+}

--- a/tests/errors/comprehensive/E2002_for_extra_closing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_extra_closing_paren.ez
@@ -1,0 +1,10 @@
+/*
+ * Error Test: E2002 - for loop without opening paren but with closing paren
+ * Expected: "expected {, got ) instead"
+ */
+
+do main() {
+    for i in range(0, 3)) {
+        temp x int = i
+    }
+}

--- a/tests/errors/comprehensive/E2002_for_missing_closing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_missing_closing_paren.ez
@@ -1,0 +1,10 @@
+/*
+ * Error Test: E2002 - for loop with opening paren but missing closing paren
+ * Expected: "expected ), got { instead"
+ */
+
+do main() {
+    for (i in range(0, 3) {
+        temp x int = i
+    }
+}

--- a/tests/errors/comprehensive/E2002_for_nested_missing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_nested_missing_paren.ez
@@ -1,0 +1,12 @@
+/*
+ * Error Test: E2002 - nested for loop with mismatched parens
+ * Expected: "expected ), got { instead"
+ */
+
+do main() {
+    for (i in range(0, 3)) {
+        for (j in range(0, 2) {
+            temp x int = i + j
+        }
+    }
+}

--- a/tests/errors/comprehensive/E2002_for_with_type_missing_paren.ez
+++ b/tests/errors/comprehensive/E2002_for_with_type_missing_paren.ez
@@ -1,0 +1,10 @@
+/*
+ * Error Test: E2002 - for loop with type annotation and missing closing paren
+ * Expected: "expected ), got { instead"
+ */
+
+do main() {
+    for (i int in range(0, 3) {
+        temp x int = i
+    }
+}

